src/services/transaction_watcher_signer_service.ts

```
@@ -0,0 +1,364 @@
+        }
+        return signer;
+    }
+    private async _getNextSignerAsync(): Promise<Signer> {
+        const sortedSigners = await this._getSortedSignerPublicAddressesByAvailabilityAsync();
+        // TODO(oskar) - add random choice for top signers to better distribute
+        // the fees.
+        const signer = this._signers.get(sortedSigners[0]);
+        if (signer === undefined) {
+            throw new Error(`signer with public address: ${sortedSigners[0]} is not available`);
+        }
+
+        return signer;
+    }
+    private async _getSortedSignerPublicAddressesByAvailabilityAsync(): Promise<string[]> {
+        const map = new Map<string, number>();
+        this._availableSignerPublicAddresses.forEach(signerAddress => {
+            map.set(signerAddress, 0);
+        });
+        // TODO(oskar) - move to query builder?
+        const res: Array<{ from: string; count: number }> = await this._transactionRepository.query(
+            `SELECT transactions.from, COUNT(*) FROM transactions WHERE status in ('submitted','mempool','stuck') GROUP BY transactions.from`,
+        );
+        res.filter(result => {
+            // we exclude from addresses that are not part of the available
+            // signer pool
+            return this._availableSignerPublicAddresses.includes(result.from);
+        }).forEach(result => {
+            map.set(result.from, result.count);
+        });
+        return [...map.entries()]
+            .sort((a, b) => {
+                return a[1] - b[1];
+            })
+            .map(entry => entry[0]);
+    }
+    private async _unstickTransactionAsync(
+        tx: TransactionEntity,
+        gasPrice: BigNumber,
+        signer: Signer,
+    ): Promise<string> {
+        if (tx.nonce === undefined) {
+            throw new Error(`failed to unstick transaction ${tx.txHash} nonce is undefined`);
+        }
+        const txHash = await signer.sendTransactionToItselfWithNonceAsync(tx.nonce, gasPrice);
```

we  should wrap this in a try/catch or the caller should, as the sendTransaction can fail (no network, or invalid params)

---

We catch it in the encompassing `_checkForStuckTransactionsAsync()` already, and it will be retried in the next call to it.

I guess we could wrap this in a try/catch to retry the sending inline?